### PR TITLE
Added BlendTree to right click create asset menu.

### PR DESCRIPTION
**Pull request by rurre**
Not sure where to put it, so into the root editor folder it goes.

### CODE REVIEW

File: Editor/CreateBlendTree.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+using System.IO;
+using UnityEditor;
+using UnityEditor.Animations;
+using UnityEditor.ProjectWindowCallback;
+using UnityEngine;
+
+namespace nadena.dev.modular_avatar.core.editor
+{
+    public class CreateBlendTree : Editor
```
Comment (bdunderscore) [inline Editor/CreateBlendTree.cs:9]: Please make this an `internal static class` as this shouldn't be an exposed API.

File: Editor/CreateBlendTree.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {
+        [MenuItem("Assets/Create/BlendTree", priority = 411)]
+        static void CreateNewBlendTree()
+        {
+            ProjectWindowUtil.StartNameEditingIfProjectWindowExists(
```
Comment (bdunderscore) [inline Editor/CreateBlendTree.cs:14]: Is this an undocumented API?

File: Editor/CreateBlendTree.cs (hunk 1 above)
Comment (rurre) [reply Editor/CreateBlendTree.cs]: Done.

File: Editor/CreateBlendTree.cs (hunk 2 above)
Comment (rurre) [reply Editor/CreateBlendTree.cs:14]: It's seemingly an older API. Seems to be missing from later unity version documentation. There's a page on end rename action, but not start.
https://docs.unity3d.com/ScriptReference/ProjectWindowCallback.EndNameEditAction.Action.html

But it is in use by default assets
https://github.com/Unity-Technologies/UnityCsReference/blob/496f7d6c5c0882f35bb776e96356712d54710033/Editor/Mono/ProjectBrowser/ProjectWindowUtil.cs#L347

What this ends up doing is beginning to rename a new asset, confirming the name then actually creating it, like any default asset you create.

**Comment by bdunderscore** [comment]
I'll look to merging this when I have the blend tree merge component implemented.

**Pull request merged**